File: backend/internal/domain/attendance.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type AttendanceStatus string

const (
	AttendancePresent AttendanceStatus = "present"
	AttendanceAbsent  AttendanceStatus = "absent"
	AttendanceLate    AttendanceStatus = "late"
	AttendanceExcused AttendanceStatus = "excused"
)

type Attendance struct {
	ID         uuid.UUID        `db:"id"`
	GroupID    uuid.UUID        `db:"group_id"`
	StudentID  uuid.UUID        `db:"student_id"`
	TeacherID  uuid.UUID        `db:"teacher_id"`
	LessonDate time.Time        `db:"lesson_date"`
	Status     AttendanceStatus `db:"status"`
	Note       *string          `db:"note"`
	CreatedAt  time.Time        `db:"created_at"`
	UpdatedAt  time.Time        `db:"updated_at"`
}

File: flutter_app/lib/features/attendance/domain/entities/attendance.dart
import 'package:flutter/foundation.dart';

enum AttendanceStatus {
  present,
  absent,
  late,
  excused;

  static AttendanceStatus fromJson(String v) => switch (v) {
        'absent' => absent,
        'late' => late,
        'excused' => excused,
        _ => present,
      };

  String toJson() => name;

  String get label => switch (this) {
        present => 'Присутствует',
        absent => 'Отсутствует',
        late => 'Опоздал',
        excused => 'Уважительная',
      };
}

@immutable
class Attendance {
  const Attendance({
    required this.id,
    required this.groupId,
    required this.studentId,
    required this.teacherId,
    required this.lessonDate,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.note,
  });

  final String id;
  final String groupId;
  final String studentId;
  final String teacherId;
  final String lessonDate; // "2024-09-01"
  final AttendanceStatus status;
  final String? note;
  final DateTime createdAt;
  final DateTime updatedAt;
}
